packages/edit-post/src/editor.js

```diff
@@ -56,13 +56,13 @@ function Editor( {
 				select( coreStore );
 			const { getEditorSettings } = select( editorStore );
 			const { getBlockTypes } = select( blocksStore );
-			const isTemplate = [ 'wp_template', 'wp_template_part' ].includes(
+			const isTemplatePart = [ 'wp_template_part' ].includes(
 				currentPost.postType
 			);
 			// Ideally the initializeEditor function should be called using the ID of the REST endpoint.
 			// to avoid the special case.
 			let postObject;
-			if ( isTemplate ) {
+			if ( isTemplatePart ) {
```

Comment:
I'm wondering about the reasoning for this change. I don't really know what the code does though. 😬

Reply:
Previously the template of the current post was retrieved using a call to `getEntityRecords` and the following in the params to this call
```
{
    wp_id: currentPost.postId,
}
```
but now when switching to edit the template the currentPost.postId is actually the template id, so the above call would fail as it is not recognised as a `wp_id`, so that branch is now only relevant to template parts  and we can instead  get the template with the branch after this which uses :
```
postObject = getEntityRecord(
	'postType',
	currentPost.postType,
	currentPost.postId
);
```

Reply:
Ok, makes sense. 👍 

The code can be simplified to `if ( currentPost.postType === 'wp_template_part' )` rather than using the array.includes method.

Reply:
> Previously the template of the current post was retrieved using a call to getEntityRecords and the following in the params to this call

Actually, this doesn't seem to be true. In my testing in `trunk`, `isTemplatePart` never seems to be `true` in the course of editing a post or its associated template. `currentPost.postType` represents the original edited post type and never the template, so there would be no way for it to be true.

The change does make sense, but it leaves me to wonder what the truthy statement is for, and whether the change might be breaking something we don't know about.

Reply:
Looks like there is an issue there with autosaves now when editing a templates, which #27910 fixed:
<img width="916" alt="Screenshot 2024-01-22 at 2 51 04 PM" src="https://github.com/WordPress/gutenberg/assets/3629020/e4aaea8f-761c-46ec-a351-1b4bdef8fad8">

so will need to take another look at this change.

Reply:
@talldan, like you I could not find any instances when the truthy branch is used. I also went back to trunk prior to the comment when we introducted the `usePostHistory` hook and when navigating from post to templates this is never fired, because the post id and type always remain as the current post being edited. I have removed this branc for now. @riad, can you see any remaining need for [this check ](https://github.com/WordPress/gutenberg/blob/trunk/packages/edit-post/src/editor.js#L59) that we are overlooking?